script.js

```diff
@@ -9,89 +9,206 @@
 // Once he guesses it correctly, end the loop.
 // Hint: What's the difference between a do-while loop and a while loop?
 
+// do {
 
+//     var randomNumber = Math.floor(Math.random()*10) + 1;
 
+//     var userGuess = parseInt(prompt("Enter an integer between 1 to 10"));
 
+//     if (userGuess === randomNumber) {
 
+//         console.log("You've guessed it! The random number was: " + randomNumber);
 
+//     } else {
+
+//         console.log("Wrong number! The random number was: " + randomNumber);
+
+//     }
+
+// } while (userGuess !== randomNumber);
 
 // Exercise 2: Reversing a String
 // ==============================
 // Write a function that takes a string as an argument and returns its reverse. For example, reverseString("abc defg")
 // should return "gfed cba".
 
 
+// var reverseString = function(string) {
+
+//     var splitString = string.split("");
 
+//     var reverseArray = splitString.reverse();
 
+//     var joinArray = reverseArray.join("");
 
+//     return joinArray;
+// }
 
+// console.log(reverseString(prompt("Input a string to reverse")));
 
 
 // Exercise 3
 // ==========
-var numArray = [3, 9, 2, 5, 3, 6, 7, 4, 8, 1, 4, 10, 21, 43, 56, 23, 54, 94, 94, 0, -2, 4, 2, 7];
+// var numArray = [3, 9, 2, 5, 3, 6, 7, 4, 8, 1, 4, 10, 21, 43, 56, 23, 54, 94, 94, 0, -2, 4, 2, 7];
 
 // Write functions that each take an array of numbers, e.g. numArray, and...
 // 1. return an array with the elements in reverse order, e.g. reverseArray([1,2,3]) should return [3,2,1]
+
+// var reverseArray = function(numArray) {
+
+//     var newNumArray = new Array;
+
+//     for (var i = numArray.length-1; i >= 0; i--) {
+
+//         newNumArray.push(numArray[i]);
+
+//     }
+
+//     return newNumArray;
+
+// }
+
+// console.log(reverseArray(numArray));
+
 // 2. return an array with each element of the input array multiplied by 2, e.g. double([1,2,3]) should return [2,4,6]
+
+// var reverseArray = function(numArray) {
+
+//     var newNumArray = new Array;
+
+//     for(var i = numArray.length-1; i >= 0; i--) {
+
+//         newNumArray.push(numArray[i] * 2);
+
+//     }
+
+//     return newNumArray;
+
+// }
+
+// console.log(reverseArray(numArray));
+
 // 3. return the sum up all the elements of the input array, e.g. sum([1,2,3]) should return 6
-// 4. return an array with only odd numbers from the input array, e.g. onlyOdd([1,2,3,4,5]) should return [1,3,5]
-// 5. (Bonus) return an array where each element in this new array is the sum of the element before it and the element in its current position in the old array, e.g. the first few elements in this new array would be `[3, 12, 11, 7, 8, ...]`
 
+// Option 1:
+// var sumOfArray = numArray.reduce((a, b) => a + b, 0);
```

Comment:
Interestingly, if you've learnt about Array.prototype.reduce(…), why not try to do something similar for the other parts of this exercise? They are all modeled after similar array methods, e.g. map, and filter. ;)